utils/model_uploader/model_autotracing.py

```diff
@@ -0,0 +1,488 @@
+                model_id=model_id, sentences=TEST_SENTENCES
+            )
+        else:
+            model_path = pre_trained_model.save_as_onnx(model_id=model_id)
+    except Exception as e:
+        assert False, f"Raised Exception during saving model as {model_format}: {e}"
+
+    # 3.) Create a model config json file
+    try:
+        pre_trained_model.make_model_config_json(
+            version_number=model_version,
+            model_format=model_format,
+            embedding_dimension=embedding_dimension,
+            pooling_mode=pooling_mode,
+        )
+    except Exception as e:
+        assert (
+            False
+        ), f"Raised Exception during making model config file for {model_format} model: {e}"
+
+    # 4.) Return model_path & model_config_path for model registration
+    model_config_path = folder_path + MODEL_CONFIG_FILE_NAME
+    return model_path, model_config_path
+
+
+def register_and_deploy_sentence_transformer_model(
+    ml_client: "MLCommonClient",
+    model_path: str,
+    model_config_path: str,
+    model_format: str,
+) -> List["DTypeLike"]:
+    """
+    Register the pretrained sentence transformer model by using the model file and the model config file,
+    deploy the model to generate embeddings for the TEST_SENTENCES,
+    and return the embeddings for model verification
+
+    :param ml_client: A client that communicates to the ml-common plugin for OpenSearch
+    :type ml_client: MLCommonClient
+    :param model_path: Path to model file
+    :type model_path: string
+    :param model_config_path: Path to model config file
+    :type model_config_path: string
+    :param model_format: Model format ("TORCH_SCRIPT" or "ONNX")
+    :type model_format: string
+    """
+    embedding_data = None
+
+    # 1.) Register the model
+    model_id = ""
+    task_id = ""
+    try:
+        model_id = ml_client.register_model(
+            model_path=model_path,
+            model_config_path=model_config_path,
+            deploy_model=False,
+            isVerbose=True,
+        )
+        print()
+        print(f"{model_format}_model_id:", model_id)
+        assert model_id != "" or model_id is not None
+    except Exception as e:
+        assert False, f"Raised Exception in {model_format} model registration: {e}"
+
+    # 2.) Deploy the model
+    try:
+        ml_client.deploy_model(model_id)
+        api_url = f"{ML_BASE_URI}/models/{model_id}/_deploy"
+        task_id = ml_client._client.transport.perform_request(
+            method="POST", url=api_url
+        )["task_id"]
+        assert task_id != "" or task_id is not None
+        ml_model_status = ml_client.get_model_info(model_id)
+        assert ml_model_status.get("model_state") != "DEPLOY_FAILED"
+        print(f"{model_format}_task_id:", task_id)
+    except Exception as e:
+        assert False, f"Raised Exception in {model_format} model deployment: {e}"
+
+    # 3.) Check model status
+    try:
```

Comment:
We don't need to check model task status also. IF model is deployed then we can start generating embedding.